rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -2064,19 +2069,41 @@ extern void ppu_finalize(const ppu_module& info)
 #endif
 }
 
-extern void ppu_precompile(std::vector<std::string>& dir_queue)
+extern void ppu_precompile(std::vector<std::string>& dir_queue, std::vector<lv2_prx*>* loaded_prx)
 {
-	std::vector<std::pair<std::string, u64>> file_queue;
-	file_queue.reserve(2000);
+	// Remove duplicates
+	std::sort(dir_queue.begin(), dir_queue.end());
+	dir_queue.erase(std::unique(dir_queue.begin(), dir_queue.end()), dir_queue.end());
 
-	// Initialize progress dialog
-	g_progr = "Scanning directories for SPRX libraries...";
+	const std::string firmware_sprx_path = vfs::get("/dev_flash/sys/external/");
+
+	// Map fixed address executables area, fake overlay support
+	const bool had_ovl = !vm::map(0x3000'0000, 0x1000'0000, 0x200).operator bool();
+	const u32 ppc_seg = std::exchange(g_ps3_process_info.ppc_seg, 0x3);
```

Comment:
Doesn't seem to have any effect, what does it mean? In other place the only allowed value is 1.

Reply:
0x3 is both overlay flag and something called "fixed address PRX" which is not supported by rpcs3 atm. In the future there may be added ppc_seg checks in ppu_load_overlay and stuff like that, so it's for future proofing.